### PR TITLE
Remove virtual from ToBytes method

### PR DESCRIPTION
**Pull request by JoshLove-msft**
No need to mock the method call as it is not a service call.

### CODE REVIEW

File: sdk/core/Azure.Core.Amqp/src/AmqpAnnotatedMessage.cs [hunk 1]
```diff
@@ -161,7 +161,7 @@ section switch
         /// <seealso href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#section-message-format"/>
         /// </summary>
         /// <returns>A <see cref="BinaryData"/> instance representing the serialized message.</returns>
-        public virtual BinaryData ToBytes()
+        public BinaryData ToBytes()
```
Comment (jsquire) [inline sdk/core/Azure.Core.Amqp/src/AmqpAnnotatedMessage.cs:164]: We're not concerned with testing/mocking here?   That seems odd.

Comment (JoshLove-msft) [reply sdk/core/Azure.Core.Amqp/src/AmqpAnnotatedMessage.cs:164]: Well mocking is not needed as there are no service calls. Customers can just create an instance of the message and let the serialization do its thing. Is there some scenario where you expect customers to want to mock?

**Review by jsquire — APPROVED** [review]
Approving to unblock, probably good to get architect feedback on whether it's needed.

**Comment by JoshLove-msft** [comment]
> Approving to unblock, probably good to get architect feedback on whether it's needed.

/cc @KrzysztofCwalina @tg-msft thoughts?

**Comment by JoshLove-msft** [comment]
> > Approving to unblock, probably good to get architect feedback on whether it's needed.
> 
> /cc @KrzysztofCwalina @tg-msft thoughts?



> > Approving to unblock, probably good to get architect feedback on whether it's needed.
> 
> /cc @KrzysztofCwalina @tg-msft thoughts?

Confirmed that it is correct to leave off virtual here since there are public methods to create the type.

**Pull request merged**